flocker/node/agents/blockdevice.py

```diff
@@ -719,13 +736,20 @@ def run(self, deployer):
         except:
             return fail()
 
-        volume = api.create_volume(
-            dataset_id=UUID(self.dataset.dataset_id),
-            size=allocated_size(
-                allocation_unit=api.allocation_unit(),
-                requested_size=self.dataset.maximum_size,
-            ),
-        )
+        profile_name = self.dataset.metadata.get(PROFILE_METADATA_KEY)
+        dataset_id = UUID(self.dataset.dataset_id)
+        size = allocated_size(allocation_unit=api.allocation_unit(),
+                              requested_size=self.dataset.maximum_size)
+        if profile_name:
+            volume = (
+                deployer.profiled_blockdevice_api.create_volume_with_profile(
+                    dataset_id=dataset_id,
+                    size=size,
+                    profile_name=profile_name
+                )
+            )
+        else:
+            volume = api.create_volume(dataset_id=dataset_id, size=size)
```

Comment:
It might help readability a bit to factor all of this `create_volume` related code out into a private helper method, I think.